### ISSUE TITLE
`//src/test/py/bazel:py_test` is broken with Python 3.11

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#20660]
https://buildkite.com/bazel/bazel-bazel/builds/26029#018c915e-33cc-4272-9169-2722a7a94f6d 

We recently update our mac machines on CI and python was upgraded to 3.11. This test started to fail with

```
File "/private/var/tmp/_bazel_buildkite/42e36ea1c1dcd25f4ab5af8d98bf3465/sandbox/darwin-sandbox/8220/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/_bazel_buildkite/da87a3ce985221b3cb431edb4e3cae38/execroot/_main/bazel-out/darwin_x86_64-fastbuild/bin/a/a.runfiles/_main/a/a.py", line 1, in <module>
import b
ModuleNotFoundError: No module named 'b'
```

To reproduce on macOS
```
brew install python3
bazel test //src/test/py/bazel:py_test --test_filter=PyTest.testSmoke
```

**Comment by meteorcloudy** [comment]
I debugged a bit with the `sys.path` printed

with python 3.9 as default python on PATH
```
ci@bk-imacpro-11 tmpdp3x5ecv % ./bazel-bin/a/a
['/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/a', '/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles', '/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles/_main', '/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles/bazel_tools', '/Applications/Xcode14.2.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python39.zip', '/Applications/Xcode14.2.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9', '/Applications/Xcode14.2.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/lib-dynload', '/Applications/Xcode14.2.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/site-packages']
Hello, World
```

with python 3.11 as default python on PATH
```
ci@bk-imacpro-11 tmpdp3x5ecv % ./bazel-bin/a/a
['/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles', '/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles/_main', '/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/bazel-bin/a/a.runfiles/bazel_tools', '/usr/local/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python311.zip', '/usr/local/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11', '/usr/local/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload', '/usr/local/lib/python3.11/site-packages']
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_ci/ba891f1787a15980e5a85b9711097c39/sandbox/darwin-sandbox/3558/execroot/_main/_tmp/9d5b13726e352fe4fa73b514c6621472/tests_root/tmpdp3x5ecv/./bazel-bin/a/a.runfiles/_main/a/a.py", line 3, in <module>
    import b
ModuleNotFoundError: No module named 'b'
```

**Comment by meteorcloudy** [comment]
I'm disabling this test case on macOS to make presubmit green

**Comment by rickeylev** [comment]
This is probably due to `PYTHONSAFEPATH` being respected in python 3.11 (it isn't in earlier python versions). This is a good thing and is just exposing a non-hermetic behavior that was being relied upon.

Should be a fairly simple fix. Either the import line needs to change ("from something import x") or an `imports = something` setting needs to be added a py_* target somewhere

**Comment by rickeylev** [comment]
Looks like a simple fix. I have https://github.com/bazelbuild/bazel/pull/20738 going through CI.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.5.0 RC2](https://github.com/bazelbuild/bazel/releases/tag/7.5.0rc2). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.5.0rc2. Thanks!
